packages/expo-notifications/android/src/main/java/expo/modules/notifications/tokens/interfaces/PushTokenManager.java

```diff
@@ -19,4 +19,11 @@
    *                 with {@link PushTokenManager#addListener(PushTokenListener)}.
    */
   void removeListener(PushTokenListener listener);
+
+  /**
+   * Notifies listeners of a new token.
+   *
+   * @param token New device push token.
+   */
+  void onNewToken(String token);
```

Comment:
Good thinking! However, if that isn't too troublesome (I could do it if you want!), I think it would be better if we created a separate interface just for this function — the `PushTokenManager` is supposed to be an interface between Push Token Manager Singleton Module and Push Token Exported Module and the exported module shouldn't know about the manager having a `onNewToken` method.

Let's create a `FirebaseTokenListener` interface with `void onNewToken(String token);` and use it throughout `FirebaseListenerService` whenever `PushTokenManager` is registered. 🙏

Reply:
makes sense! I've added a new `FirebaseTokenListener` class in the same tokens.interfaces package, but let me know if you think it would make more sense elsewhere.